### PR TITLE
Add more options to verify

### PR DESCRIPTION
**Pull request by vthapar**
1. Replaces `verify-conectivity` with `verify`
2. Adds following options to `verify`
   a. `connectivity` - true by default, runs submariner e2e
   b. `service-discovery` - false, runs lighthouse e2e
   c. `all` - runs both `connectivity` and `servicediscovery` tests

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,6 +18,7 @@ require (
 	github.com/prometheus/client_golang v1.2.1 // indirect
 	github.com/spf13/cobra v0.0.7
 	github.com/spf13/pflag v1.0.5
+        github.com/submariner-io/lighthouse v0.3.1-0.20200521062404-6d5bf9939fef
```
Comment (mangelajo) [inline go.mod:21]:
```suggestion
       github.com/submariner-io/lighthouse v0.3.1-0.20200521062404-6d5bf9939fef
```

Comment (vthapar) [reply go.mod:21]: Locally it shows fine for me but in PR it shows up with extra space.

File: pkg/subctl/cmd/verify.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	verifyconnectivity "github.com/submariner-io/submariner-operator/pkg/subctl/verify/connectivity"
+	verifyservicediscovery "github.com/submariner-io/submariner-operator/pkg/subctl/verify/servicediscovery"
+)
+
+var (
+	verifyConnectivity     bool
+	verifyServiceDiscovery bool
+	verifyAll              bool
+	//verboseConnectivityVerification bool
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:35]: Can we remove those comments? see my later comment about refactor.

File: pkg/subctl/cmd/verify.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	verifyCmd.Flags().BoolVar(&verboseConnectivityVerification, "verbose", false, "Produce verbose logs during connectivity verification")
+	verifyCmd.Flags().UintVar(&operationTimeout, "operation-timeout", 240, "Operation timeout for K8s API calls")
+	verifyCmd.Flags().UintVar(&connectionTimeout, "connection-timeout", 60, "The timeout in seconds per connection attempt ")
+	verifyCmd.Flags().UintVar(&connectionAttempts, "connection-attempts", 2, "The maximum number of connection attempts")
+	verifyCmd.Flags().StringVar(&reportDirectory, "report-dir", ".", "XML report directory")
+
+	rootCmd.AddCommand(verifyCmd)
+}
+
+var verifyCmd = &cobra.Command{
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:55]: can we MarkDeprecate the other verify-connectivity command? 

https://godoc.org/github.com/spf13/pflag#FlagSet.MarkDeprecated

File: pkg/subctl/cmd/verify.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	//connectionTimeout               uint
+	//connectionAttempts              uint
+	//reportDirectory                 string
+)
+
+func init() {
+	verifyCmd.Flags().BoolVar(&verifyConnectivity, "connectivity", true, "verify connectivity between two clusters")
+	verifyCmd.Flags().BoolVar(&verifyServiceDiscovery, "service-discovery", false, "verify service-discovery between two clusters")
+	verifyCmd.Flags().BoolVar(&verifyAll, "all", false, "verify connectivity and service-discovery between two clusters")
+	verifyCmd.Flags().BoolVar(&verboseConnectivityVerification, "verbose", false, "Produce verbose logs during connectivity verification")
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:46]: Let's move the common flags to a function:

https://github.com/submariner-io/submariner-operator/blob/master/pkg/subctl/cmd/join.go#L77

like we do in the subctl/cmd/broker.go

Comment (vthapar) [reply pkg/subctl/cmd/verify.go:46]: Will do.

File: pkg/subctl/cmd/verify.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	Use:   "verify <kubeConfig1> <kubeConfig2>",
+	Short: "Verify connectivity/service-discovery between two clusters",
+	Args: func(cmd *cobra.Command, args []string) error {
+		return checkValidateArguments(args)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		configureTestingFramework(args)
+		verifyConnectivity = verifyConnectivity || verifyAll
+		verifyServiceDiscovery = verifyServiceDiscovery || verifyAll
+		if verifyConnectivity {
```
Comment (skitt) [inline pkg/subctl/cmd/verify.go:65]:
```suggestion
		if verifyConnectivity || verifyAll {
```

File: pkg/subctl/cmd/verify.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+			verifyconnectivity.RunE2E(&testing.T{})
+		}
+		if verifyServiceDiscovery {
```
Comment (skitt) [inline pkg/subctl/cmd/verify.go:68]:
```suggestion
		if verifyServiceDiscovery || verifyAll {
```

Comment (vthapar) [reply pkg/subctl/cmd/verify.go]: Done.

File: pkg/subctl/cmd/verify.go (hunk 5 above)
Comment (vthapar) [reply pkg/subctl/cmd/verify.go]: Done

File: pkg/subctl/cmd/verify.go (hunk 3 above)
Comment (vthapar) [reply pkg/subctl/cmd/verify.go:59]: Done

File: pkg/subctl/cmd/verify.go (hunk 2 above)
Comment (vthapar) [reply pkg/subctl/cmd/verify.go]: Done

File: pkg/subctl/cmd/verify.go (hunk 3 above)
Comment (mangelajo) [reply pkg/subctl/cmd/verify.go:59]: Thanks

File: pkg/subctl/cmd/verify.go [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+func init() {
+	addVerifyFlags(verifyCmd)
+	rootCmd.AddCommand(verifyCmd)
+}
+
+func addVerifyFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&verifyConnectivity, "connectivity", true, "verify connectivity between two clusters")
+	cmd.Flags().BoolVar(&verifyServiceDiscovery, "service-discovery", false, "verify service-discovery between two clusters")
+	cmd.Flags().BoolVar(&verifyAll, "all", false, "verify connectivity and service-discovery between two clusters")
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:50]: Those three are particular to this command only, not the old deprecated one, can we move those 3 to the init?

Comment (vthapar) [reply pkg/subctl/cmd/verify.go:50]: I thought I did that, likely forgot to commit before pushing and cleaned it up. will do.

File: scripts/kind-e2e/e2e.sh [hunk 8]
```diff
@@ -49,19 +35,16 @@ run_subm_clusters verify_subm_deployed
 echo "Running subctl a second time to verify if running subctl a second time works fine"
 with_context cluster3 subctl_install_subm
 
-with_context cluster3 deploy_resource "${RESOURCES_DIR}/nginx-demo.yaml"
-[[ "${lighthouse}" = "true" ]] && with_context cluster3 kubectl apply -f "${DAPPER_SOURCE}/scripts/kind-e2e/nginx-demo-export.yaml"
-with_context cluster2 deploy_resource "${RESOURCES_DIR}/netshoot.yaml"
+# dataplane E2E need to be modified for globalnet
+verify="--connectivity"
+[[ "$lighthouse" != "true" ]] || verify="--all"
```
Comment (mangelajo) [inline scripts/kind-e2e/e2e.sh:40]:
```suggestion
# dataplane E2E need to be modified for globalnet
if [[ "$lighthouse" == "true" ]]; then
    verify="--all"
else
     verify="--connectivity"
fi
```

Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh:40]: I know it's longer, but easier to reason about ':D

**Pull request merged**